### DISCUSSION TITLE
Collision planning inconsistency when using cuRobo python library with ROS 2

### DISCUSSION THREAD

**Discussion by shengyangzhuang** [NVlabs/curobo#324] [Software Q&A] [answered]
Dear cuRobo developers,

Thank you very much for the fantastic paper.

I am using cuRobo to generate collision-free path and using joint-space control method to command KUKA iiwa7 robot with [LBR-stack](https://github.com/lbr-stack/lbr_fri_ros2_stack).

I used the usd file to visualize the collision-free path, which worked well. However, when I replicated the obstacle setting in gazebo and tested path planning with cuRobo, the robot would still hit the obstacles. What will be the possible reasons for this?

Thank you very much in advance for your assistance.

With warm regards,

shengyangzhuang

https://github.com/NVlabs/curobo/assets/100099974/39d7a106-fdc6-46f1-83bd-ebc2eba223df

https://github.com/NVlabs/curobo/assets/100099974/877b8b78-2d13-4a5a-8074-d8fd6f7a4aed

**Comment by balakumar-s** [comment, marked as answer]
Your PD controller in simulation might not be tracking the trajectory accurately. This might be useful: https://github.com/NVlabs/curobo/discussions/285

**Reply by shengyangzhuang** [reply]
Thank you very much. It's very helpful. I will have a look and update in due time :)
